updateweight2.py

```diff
@@ -0,0 +1,174 @@
+NOTE_WEIGHT_SWING_FACTOR = 0.5
+
+STATUS_SKIPPED = 0
+STATUS_SUCCESSFUL = 1
+STATUS_ERROR = 2
+STATUS_KEYBOARD_INTERRUPT = 3
+STATUS_EXIT = 4
+
+NUMBER_OF_PROCESSES = 20
+
+FILENAME_SUCCESSFUL_LOG = 'success.json'
+FILENAME_SONG_LIST_JSON = 'newsongsjson.txt'
+
+difficultyKeys = ['easy', 'normal', 'hard', 'expert', 'master']
+
+TITLE_WELCOME = '''# Welcome to LLHelper LLSIF Song Updater!
+
+* Song data based on `%s`.
+* Program originally authored by **Chazeon** and the end of Sept 2017 in NY.
+
+* We are now updating song list file: `%s`.
+* Song log will be loaded from and written to: `%s`.
+''' % (LLSIF_WIN_API_DOMAIN, FILENAME_SONG_LIST_JSON, FILENAME_SUCCESSFUL_LOG)
+
+class Note:
+    def __init__(self, noteDict):
+        self.level = noteDict['notes_level']
+        self.type = noteDict['effect']
+        self.position = noteDict['position']
+        self.attribute = noteDict['notes_attribute']
+        self.appearTime = noteDict['timing_sec']
+        self.effectValue = noteDict['effect_value']
+    def isHold(self):
+        return self.type in [NOTE_TYPE_HOLD, NOTE_TYPE_SWING_HOLD]
+    def isSwing(self):
+        return self.type in [NOTE_TYPE_SWING, NOTE_TYPE_SWING_EVENT, NOTE_TYPE_SWING_HOLD]
+    def getNoteWeightedValue(self):
+        weightValue = NOTE_WEIGHT_BASE
+        if self.isSwing():
+            weightValue *= NOTE_WEIGHT_SWING_FACTOR
+        if self.isHold():
+            weightValue *= NOTE_WEIGHT_HOLD_FACTOR
+        return weightValue
+
+def getLiveMapJsonUrl(liveId):
+    return LLSIF_WIN_API_DOMAIN + LLSIF_WIN_API_ENDPOINT + str(liveId)
+
+def getLiveMap(liveId):
+    liveJsonUrl = getLiveMapJsonUrl(liveId)
+    liveJsonFp = urllib2.urlopen(liveJsonUrl)
+    liveJson = json.load(liveJsonFp)
+    return liveJson
+
+def getPositionWeight(liveId):
+    liveMap = getLiveMap(liveId)
+    positionWeight = [0.0] * 9
+    for note in liveMap:
+        note = Note(note)
+        position = 9 - note.position
+        positionWeight[position] += note.getNoteWeightedValue()
+    return map(str, positionWeight)
+
+def updatePositionWeight(live):
+    liveId = int(live['liveid'])
+    try:
+        result = getPositionWeight(liveId)
+        return (liveId, None, STATUS_SUCCESSFUL, '* Successfully processed %d' % (liveId), result)
+    except KeyboardInterrupt as e:
+        return (liveId, e, STATUS_KEYBOARD_INTERRUPT, '* User trying to exit program', None)
+    except Exception as e:
+        return (liveId, e, STATUS_ERROR, '* Failed to process %d' % (liveId), None)
+
+def positionWeightUpdateWorker(liveQueue, messageQueue, pipeOut):
+    while True:
+        if pipeOut.poll():
+            signal = pipeOut.recv()
+            if signal == STATUS_EXIT: break
+        if liveQueue.empty(): break
+        live = liveQueue.get()
+        messageQueue.put(updatePositionWeight(live))
+
+def main():
+    songs = {}
+    prevLoadedLives = []
+    updatedLives = []
+
+    messageQueue = multiprocessing.Queue()
```

Comment:
Do we really want to multi-process the fetching process? Every update contains only a few new maps and multithreaded code is super hard to read and maintain.